frontend/src/pages/pipelines/global/runs/ManageRunsTable.tsx

```diff
@@ -0,0 +1,173 @@
+import React from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import { Button, Flex, ToolbarGroup, ToolbarItem, Tooltip } from '@patternfly/react-core';
+import { TableVariant } from '@patternfly/react-table';
+
+import { TableBase, getTableColumnSort, useCheckboxTable } from '~/components/table';
+import DashboardEmptyTableView from '~/concepts/dashboard/DashboardEmptyTableView';
+import { pipelineRunColumns } from '~/concepts/pipelines/content/tables/columns';
+import PipelineRunTable from '~/concepts/pipelines/content/tables/pipelineRun/PipelineRunTable';
+import PipelineRunTableRow from '~/concepts/pipelines/content/tables/pipelineRun/PipelineRunTableRow';
+import PipelineRunTableToolbar, {
+  FilterProps,
+} from '~/concepts/pipelines/content/tables/pipelineRun/PipelineRunTableToolbar';
+import { FilterOptions } from '~/concepts/pipelines/content/tables/usePipelineFilter';
+import { ExperimentKFv2, PipelineRunKFv2 } from '~/concepts/pipelines/kfTypes';
+import { experimentsCompareRunsRoute } from '~/routes';
+import { usePipelinesAPI } from '~/concepts/pipelines/context';
+
+type ManageRunsTableProps = Omit<React.ComponentProps<typeof PipelineRunTable>, 'runType'> & {
+  filterProps: FilterProps;
+  selectedRunIds: string[];
+  experiment: ExperimentKFv2 | null;
+};
+
+export const ManageRunsTable: React.FC<ManageRunsTableProps> = ({
+  runs,
+  experiment,
+  selectedRunIds,
+  loading,
+  totalSize,
+  page,
+  pageSize,
+  setPage,
+  setPageSize,
+  filterProps,
+  ...tableProps
+}) => {
+  const navigate = useNavigate();
+  const [isSelectAllChecked, setIsSelectAllChecked] = React.useState(false);
+  const { namespace } = usePipelinesAPI();
+  const pageRunIds = runs.map(({ run_id: runId }) => runId);
+
+  const { tableProps: checkboxTableProps, toggleSelection } = useCheckboxTable(pageRunIds);
+  const selectedIdsRef = React.useRef<string[]>(selectedRunIds);
```

Comment:
useCheckboxTable manages selections and select all, why are you managing them outside the hook?

Reply:
Because the checkbox selections otherwise get reset between page changes.

Reply:
i tinkered around. you can just pass a default set of strings to the usecvheckboxtable hook and then set that to selections right away. this means you dont need the custom select all, on toggle, and use ref selections

Reply:
What default set of strings would one pass to useCheckboxTable? Set what to selections right away?

Reply:
that table has a prop for the selectedRunIds. and the state inside the checkbox table hook can eb set with an additional optional input param for defaults

Reply:
I think your last sentence is suggesting updating the useCheckboxTable hook with a new param for default IDs set in state, but I'm not seeing how that would result in the same table behavior as the current implementation. Have you tested this and everything behaves the same way?

Reply:
i have. you will also need to then use the selections from the checkbox table hook

Reply:
Those "selections" from the checkbox hook are reset between page changes. Maybe you could elaborate.

Reply:
I asked this question to Yan last week: https://www.figma.com/proto/etbN0ccCcZR1qKCpJXQNAg?node-id=1750-107510#751898481. So there's a different expectation here it seems.

Reply:
pagination

Reply:
when i tested it on my end the selection seems to work as expected across pagination